lambdas/es/indexer/index.py

```diff
@@ -108,8 +117,107 @@ def infer_extensions(key, ext):
     return ext
 
 
-def get_contents(bucket, key, ext, *, etag, version_id, s3_client, size):
-    """get the byte contents of a file"""
+@retry(
+    stop=stop_after_attempt(MAX_RETRY),
+    wait=wait_exponential(multiplier=2, min=4, max=30),
+    retry=(retry_if_exception(should_retry_exception))
+)
+def select_manifest_meta(s3_client, bucket: str, key: str):
+    """use s3 select to quickly extract line 1 of manifest
+        returns:
+            * first line (if found), None if not found or error
+    """
+    try:
+        raw = s3_client.select_object_content(
+            Bucket=bucket,
+            Key=key,
+            Expression=SELECT_PACKAGE_META,
+            ExpressionType="SQL",
+            InputSerialization={"JSON": {"Type": "LINES"}},
+            OutputSerialization={"JSON": {}}
+        )
+        for event in raw["Payload"]:
+            if "Records" in event:
+                data = event["Records"]["Payload"]
+                return data
```

Comment:
Probably this is unlikely, but [it can contain only partial record](https://docs.aws.amazon.com/AmazonS3/latest/API/RESTSelectObjectAppendix.html#RESTSelectObjectAppendix-responses):

> Records message: Can contain a single record, partial records, or multiple records.

So I think we have to iterate till the end [like in this example](https://botocore.amazonaws.com/v1/documentation/api/latest/reference/eventstream.html).

Reply:
OK, I'm going to factor out all the common logic with Kevin's PR (select, stream handling); but probably in a clean PR.

Reply:
Actually, done in this PR.